fix(popover): removed references to pf-m-icon class

mcoker opened the pull request:
fixes https://github.com/patternfly/patternfly/issues/5411

patternfly-build commented:
Preview: https://patternfly-pr-5412.surge.sh

thatblindgeye reviewed (commented):
Looks good. Build failure just needs to be addressed. Will re-review once a build wraps to check the preview surge.

thatblindgeye commented on src/patternfly/components/Popover/popover-title.hbs:
```diff
@@ -1,9 +1,6 @@
   {{#if popover-title--attribute}}
     {{{popover-title--attribute}}}
   {{/if}}>
-  {{#if popover-title--IsIcon}}
-    {{> popover-title-icon}}
-  {{/if}}
```
Looks like no icon is being rendered anymore in the preview.  Instead of removing lines 5-7, could we instead update it to something like:

```suggestion
{{#if popover-title-icon--type}}
  {{> popover-title-icon}}
{{/if}}
```

mattnolting commented on src/patternfly/components/Popover/popover-title.hbs:
```diff
@@ -1,9 +1,6 @@
-<div class="pf-c-popover__title{{#if popover-title--IsIcon}} pf-m-icon{{/if}}{{#if popover-title--modifier}} {{popover-title--modifier}}{{/if}}"
+<div class="pf-c-popover__title{{#if popover-title--modifier}} {{popover-title--modifier}}{{/if}}"
   {{#if popover-title--attribute}}
```
Since you're having to repeat `id`, I prefer adding `id` to dedicated `.hbs` params. 

```suggestion
  {{#if popover-title--id}}
    id="{{popover-title--id}}"
  {{/if}}
  {{#if popover-title--attribute}}
```

mattnolting commented on src/patternfly/components/Popover/examples/Popover.md:
```diff
@@ -278,7 +278,7 @@ import './Popover.css'
     {{/popover-close}}
     {{#> popover-header}}
-      {{#> popover-title popover-title--IsIcon="true" popover-title--attribute='id="popover-icon-title-header"' popover-title-icon--type="bullhorn"}}
+      {{#> popover-title popover-title--attribute='id="popover-icon-title-header"' popover-title-icon--type="bullhorn"}}
```
```suggestion
      {{#> popover-title popover-title--id='popover-icon-title-header' popover-title-icon--type="bullhorn"}}
```

mattnolting reviewed (commented):
LGTM, one suggestion for `.hbs` partial

mattnolting reviewed (approved):
Beautiful 🤩 LPTM

mcoker commented:
@mattnolting @thatblindgeye used our `ternary` helper to conditionally set the `popover-title-icon--type` param value for the `--Is[Status]Alert` popovers, and also cleaned up add'l `--attribute='id="..."'` -> `'--id=""` params. This is the way those were updated, except for the card demo which I updated manually

<img width="973" alt="Screenshot 2023-03-06 at 12 45 44 PM" src="https://user-images.githubusercontent.com/35148959/223203865-0e580856-a44e-4dd7-a783-0bf7ba742b7d.png">

thatblindgeye commented on src/patternfly/components/Popover/popover-title.hbs:
```diff
@@ -1,9 +1,22 @@
+  {{#> wrapper popover-title-icon--type=(ternary popover-title-icon--type popover-title-icon--type
+        (ternary popover--IsDefaultAlert "bell"
+        (ternary popover--IsInfoAlert "info-circle"
+        (ternary popover--IsSuccessAlert "info-circle"
+        (ternary popover--IsWarningAlert "info-circle"
+        (ternary popover--IsDangerAlert "info-circle"
```
Icons appearing again! We should update the success/warning/danger icons to their original, as they're all showing up with the info icon (albeit with the status color I think)

```suggestion
        (ternary popover--IsSuccessAlert "check-circle"
        (ternary popover--IsWarningAlert "exclamation-triangle"
        (ternary popover--IsDangerAlert "exclamation-circle"
```

mcoker replied:
🤦‍♂️

mcoker commented:
@mattnolting @thatblindgeye also added a var for the icon size, and matched the title size, since we lost that when the font declarations moved to just style the title-text.

mattnolting commented:
> also added a var for the icon size

Good thinking, LPTM!

mattnolting reviewed (approved):
LPTM 🚀

patternfly-build commented:
:tada: This PR is included in version 5.0.0-alpha.28 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v5.0.0-alpha.28)
- [npm package (@alpha dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/5.0.0-alpha.28)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
